### PR TITLE
Added a simple mill wrapper script (sh)

### PR DESCRIPTION
**Pull request by lefou**
See also https://github.com/lihaoyi/mill/issues/476

### CODE REVIEW

File: mill [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+# This is a wrapper script, that automatically download mill from GitHub release pages
+# You can give the required mill version with --mill-version parameter
+# If no version is given, it falls back to the value of DEFAULT_MILL_VERSION
+
+DEFAULT_MILL_VERSION=0.2.7-10-db8525
+
+set -e
+
+if [ "x$1" = "x--mill-version" ] ; then
```
Comment (lihaoyi) [inline mill]: My Bash is a bit rusty, but why did you add the `x` prefix before the LHS and RHS of the comparison?

File: mill [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  shift
+  if [ "x$1" != "x" ] ; then
+    MILL_VERSION="$1"
+    shift
+  else
+    echo "You specified -v without a version."
```
Comment (lihaoyi) [inline mill]: Should this be `"You specified --mill-version without a version."`

File: mill [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    echo "Please provide a version that matches one provided on"
+    echo "https://github.com/lihaoyi/mill/releases"
```
Comment (lihaoyi) [inline mill]: Does this only support stable versions, or unstable versions too? It could be handy for our own development to allow a flag to easily pass in an unstable version so we can test our own code that hasn't been tagged with a numbered release

File: mill (hunk 1 above)
Comment (lihaoyi) [reply mill]: Also, to be clear, this `--mill-version` param must be the first argument, before any other parameters, and before `--interactive`/`-i`? If so we can give a nice error message if the user passes in `--mill-version` in another position, similar to what we do for `--interactive`/`-i`, by creating a dummy command-line argument in Mill's parser and error-ing out if that argument is provided. See https://github.com/lihaoyi/mill/blob/master/main/src/MillMain.scala#L94

Comment (lefou) [reply mill]: The "x$i" thing is an old habit. Can't tell you anymore, why that was needed, but 20 years ago, almost all scripts looked like this. Could be some prevention of some unwanted executions and the `test` tool, who knows?

Too keep the script simple, I parse that option only when it's the first argument. I'm a bit uncomfortable with an dummy `--mill-version` option in mill. It could also check if the given version is in fact the runtime version, as a safety measure. Then there would be no need to remove it.

File: mill (hunk 2 above)
Comment (lefou) [reply mill]: You are right, I changed it from a simple `-v` to `--mill-version` to avoid prospective collisions.

File: mill (hunk 3 above)
Comment (lefou) [reply mill]: It supports all downloadable version, also these n-commits after-release downloads. In fact, I wrote the script exactly for that use case, called it `mill-launcher` and then realized, that it's already the solution for #476.

File: mill (hunk 1 above)
Comment (lefou) [reply mill]: After a short web-search I found older/other shells could not compare an empty string properly, and some couldn't handle a `!` or `(` in `$1`. And someone said Solaris 10 still has issues.

**Comment by lefou** [comment]
I created a dedicated project for the wrapper, as the wrapped should be in general decoupled from the current mill version and also already supports older mill versions.

 https://github.com/lefou/millw
